capi/geos_c.h.in

```diff
@@ -1599,6 +1604,42 @@ extern GEOSGeometry GEOS_DLL *GEOSConvexHull(const GEOSGeometry* g);
  */
 extern GEOSGeometry GEOS_DLL *GEOSMinimumRotatedRectangle(const GEOSGeometry* g);
 
+/* Constructs the Maximum Inscribed Circle for a  polygonal geometry, up to a specified tolerance.
+ * The Maximum Inscribed Circle is determined by a point in the interior of the area
+ * which has the farthest distance from the area boundary, along with a boundary point at that distance.
+ * In the context of geography the center of the Maximum Inscribed Circle is known as the
+ * Pole of Inaccessibility. A cartographic use case is to determine a suitable point
+ * to place a map label within a polygon.
+ * The radius length of the Maximum Inscribed Circle is a  measure of how "narrow" a polygon is. It is the
+ * distance at which the negative buffer becomes empty.
+ * The class supports polygons with holes and multipolygons.
+ * The implementation uses a successive-approximation technique over a grid of square cells covering the area geometry.
+ * The grid is refined using a branch-and-bound algorithm. Point containment and distance are computed in a performant
+ * way by using spatial indexes.
+ */
+extern GEOSGeometry GEOS_DLL *GEOSMaximumInscribedCircleCenter(const GEOSGeometry* g, double tolerance);
+/* Returns a two-point linestring, with one point at the center of the inscribed circle and the other
+ * on the boundary of the inscribed circle.
+ */
+extern GEOSGeometry GEOS_DLL *GEOSMaximumInscribedCircleRadius(const GEOSGeometry* g, double tolerance);
+
+/* Constructs the Largest Empty Circle for a set of obstacle geometries, up to a
+ * specified tolerance. The obstacles are point and line geometries.
```

Comment:
Just wondering about the lack of polygon support here... Is there a particular reason why it's skipped?

Reply:
Just that it will take a bit more time to implement (it requires additional checks to restrict candidate points lying in the polygons).  Hoping to add it soon-ish.

Reply:
Fair enough!

I was mostly thinking in terms of finding the point furthest from a bunch of buildings

Reply:
Yes, that's a good use case.  Well, stay tuned...  maybe can get this added soon.

Reply:
Love it -- thanks!